### PR TITLE
Added current_user method, setup FactoryGirl config, added RSpec tests for current_user

### PR DESCRIPTION
**Pull request by baoskee**
The method current_user returns mentor instance if he is signed in, and student if he is signed in. It does not take into account cookies yet.

### CODE REVIEW

**Review by gavinksong — CHANGES_REQUESTED** [review]
You have a merge commit.

**Comment by emilioaurea** [comment]
Bao you gotta rebase bru.

File: app/controllers/conversations_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class ConversationsController < ApplicationController
-  before_action :set_conversation, only: [:show, :edit, :update, :destroy]
+  # before_action :set_conversation, only: [:show, :edit, :update, :destroy]
```
Comment (gavinksong) [inline app/controllers/conversations_controller.rb]: just delete. this is why we have git.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,17 +1,9 @@
-require "codeclimate-test-reporter"
```
Comment (gavinksong) [inline spec/spec_helper.rb:1]: Why did you delete these lines? We need these to report coverage.

File: spec/factories/student_factory.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :student do
+    email 'your_mom@berkeley.edu'
```
Comment (gavinksong) [inline spec/factories/student_factory.rb:3]: dang.

**Review by gavinksong — CHANGES_REQUESTED** [review]

File: spec/factories/student_factory.rb (hunk 3 above)
Comment (gavinksong) [reply spec/factories/student_factory.rb:3]: Bao Mai is mentoring your mom.

**Comment by baoskee** [comment]
ugh goddamnit

File: spec/spec_helper.rb (hunk 2 above)
Comment (baoskee) [reply spec/spec_helper.rb:1]: accident, oops

File: spec/factories/student_factory.rb (hunk 3 above)
Comment (baoskee) [reply spec/factories/student_factory.rb:3]: damn right

**Review by gavinksong — APPROVED** [review]
